server/cmwell-zstore/Readme.md

```diff
@@ -0,0 +1,38 @@
+# zStore
+
+zStore is a library CM-Well uses in several modules, to keep generic key-value data in a persistent (optionally temporal), distributed manner. Underlying implementation is Cassandra.
+
+zStore features are:
+- [Self contained CQL setup](#Self-contained-CQL-setup)
+- [Simple put/get API](#Simple-put/get-API)
+- [Caching layer](#Caching-layer:-zCache)
+
+## Usages
+CM-Well uses zStore for several tasks, such as
+- Keeping large FileInfotons
+- Temporarily store large payload of a large Infoton during ingest: while the meta-data is delivered from WS to BG through Kafka, if the payload is too large it is redirected to zStore and referenced from Kafka, for the BG to populate it once consumed that Kafka message.
+- TrackingActors persistence
+- STP progress persistence
+- BG/DCC progress persistence 
+
+## Features
+
+### Self contained CQL setup
+Besides `path` and `infoton` tables CM-Well has, there's another `zstore` table in the "data2" keyspace. It's columns are `uzid`, `field` and `value` where the latter is blob. zStore object runs the `CREATE IF NOT EXISTS` CQL statement when used from any module. 
+
+> Note: Cassandra recommends against running multiple "CREATE IF NOT EXISTS" statements on the same table in parallel. That can lead to schema version errors. Because of that, CM-Well initialization contains the zstore table creation, so it will be created when a cluster is created. Nonetheless, we keep the CQL statements self-contained in zStore object so it can be reused in other projects.   
+
+Part of zStore logic is to serialize / deserialize large binary values into smaller chunks in zStore table. 
+
+### Simple put/get API
+zStore provides a very simple API of put and get methods where the key is `String` and the value is `Array[Byte]`. Since this involves IO, all results are Futures. It also provides an optional TTL parameter for the put method for cases when the user wants to have the value persisted only for a certain period of time. This facilitates the built-in TTL feature in Cassandra. Please refer to the `ZStore` trait in code to see all APIs. 
```

Comment:
I don't know what "all results are Futures" means, but if you're sure the reader will, I'm OK with it.